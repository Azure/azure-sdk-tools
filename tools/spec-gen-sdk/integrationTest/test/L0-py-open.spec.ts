
import {
  createPRWithPatch, getPRListInfo, getLoggerTextForAssert
} from '../utils';
import { fixtures } from '../fixtures';
import { preparePySDKTest } from './lang-common';

describe('In open PR, SDK Automation', () => {

  it.skip('should create Python sdk pr on spec pr opened', async () => {
    const {
      specRepoName, sdkRepoName, logger, launchSDKAutomation, patchPyLog
    } = await preparePySDKTest('opened-pr');

    const specPRNumber = await createPRWithPatch(
      specRepoName, fixtures.specTest.patch0_AddService, 'patch0'
    );

    // Launch SDK Automation
    await launchSDKAutomation(specPRNumber);

    // Assert
    expect(await getPRListInfo(sdkRepoName)).toMatchSnapshot();
    expect(patchPyLog(await getLoggerTextForAssert(
      logger, specRepoName, specPRNumber, sdkRepoName
    ))).toMatchSnapshot();
  });

});
