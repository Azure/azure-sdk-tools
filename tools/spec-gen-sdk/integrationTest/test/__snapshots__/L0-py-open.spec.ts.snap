// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`In open PR, SDK Automation should create Python sdk pr on spec pr opened 1`] = `
Array [
  Object {
    "base": "release/v3",
    "head": "sdkAuto/1/azure-mgmt-testservice",
    "number": 1,
    "state": "open",
    "title": "[AutoPR azure-mgmt-testservice] patch0",
  },
]
`;

exports[`In open PR, SDK Automation should create Python sdk pr on spec pr opened 2`] = `
"github	Get specificationRepositoryConfiguration.json from <org>/<spec_repo>
info	mainRepository: <org>/<sdk_repo>
info	mainBranch: release/v3
info	integrationRepository: <org>/<sdk_repo>
info	integrationBranchPrefix: sdkAutomation
info	secondaryRepository: <org>/<sdk_repo>
info	secondaryBranch: release/v3
github	Get swagger_to_sdk_config.json from <org>/<sdk_repo>
git	Set remote main on <org>/<spec_repo> with auth
git	Delete branch refs/heads/master
git	Delete branch refs/heads/patch0
git	Fetch pull/1/merge on <org>/<spec_repo>
git	Set remote main on <org>/<sdk_repo> with auth
git	Set remote secondary on <org>/<sdk_repo> with auth
git	Set remote integration on <org>/<sdk_repo> with auth
git	Delete branch refs/heads/master
git	Delete branch refs/heads/release/v3
git	Fetch <org>/<sdk_repo> to main
git	Checkout main
git	Checkout secondary from main
warn	Skip initScript due to not configured
info	Legacy python init
command	npx n 12 
info	Config: {}
command	python3 -m venv venv 
info	Config: {}
command	pip install --upgrade setuptools wheel pip 
info	Config: {\\"logPrefix\\":\\"pip install\\"}
command	pip install pathlib jinja2 msrestazure 
info	Config: {\\"logPrefix\\":\\"pip install\\"}
info	6 changed files in spec PR:
info	A	specification/testservice/resource-manager/Microsoft.TestService/stable/2020-01-01/TestService.json
info	A	specification/testservice/resource-manager/Microsoft.TestService/stable/2020-01-01/examples/TestGet.json
info	A	specification/testservice/resource-manager/Microsoft.TestService/stable/2020-01-01/examples/TestPut.json
info	A	specification/testservice/resource-manager/readme.azureresourceschema.md
info	A	specification/testservice/resource-manager/readme.md
info	A	specification/testservice/resource-manager/readme.python.md
info	Related readme.md list:
info		specification/testservice/resource-manager/readme.md
info	Handle the following readme.md:
info		specification/testservice/resource-manager/readme.md
git	Checkout branch sdkGen
section	Call legacy generate
command	autorest --version=V2 --use=@microsoft.azure/autorest.python@4.0.73 --python --multiapi --keep-version-file --no-async --python-sdks-folder=<work_dir>/<sdk_repo>/sdk ../<spec_repo>/specification/testservice/resource-manager/readme.md
info	Config: {\\"stdout\\":{\\"showInComment\\":{}},\\"stderr\\":{\\"showInComment\\":true,\\"scriptWarning\\":{}},\\"exitCode\\":{\\"showInComment\\":true,\\"result\\":\\"error\\"}}
section	Commit generate result
git	Add * in SDK repo
info	9 changed files after SDK generate:
info	A	sdk/testservice/azure-mgmt-testservice/azure/mgmt/testservice/__init__.py
info	A	sdk/testservice/azure-mgmt-testservice/azure/mgmt/testservice/_configuration.py
info	A	sdk/testservice/azure-mgmt-testservice/azure/mgmt/testservice/_test_service_client.py
info	A	sdk/testservice/azure-mgmt-testservice/azure/mgmt/testservice/models/__init__.py
info	A	sdk/testservice/azure-mgmt-testservice/azure/mgmt/testservice/models/_models.py
info	A	sdk/testservice/azure-mgmt-testservice/azure/mgmt/testservice/models/_models_py3.py
info	A	sdk/testservice/azure-mgmt-testservice/azure/mgmt/testservice/operations/__init__.py
info	A	sdk/testservice/azure-mgmt-testservice/azure/mgmt/testservice/operations/_test_operations.py
info	A	sdk/testservice/azure-mgmt-testservice/azure/mgmt/testservice/version.py
git	Commit all the changes
section	Detect changed packages
info	Package from changed file search: /^setup.py$|^azure$/
info	1 packages found after generation:
info		sdk/testservice/azure-mgmt-testservice
git	Set remote main on <org>/<sdk_repo> with auth
git	Set remote secondary on <org>/<sdk_repo> with auth
git	Set remote integration on <org>/<sdk_repo> with auth
section	Handle package azure-mgmt-testservice
info	Package log to https://sdkautotest/dl/<org>/1/azure-sdk-for-python/azure-mgmt-testservice/logs.txt
info	baseBranch [release/v3] baseRepo [<org>/<sdk_repo>] baseRemote [integration]
info	targetBranch [sdkAuto/1/azure-mgmt-testservice] targetRepo [<org>/<sdk_repo>] baseRemote [integration]
info	Skip sync baseBranch due to same remote same branch
git	Create targetBranch sdkAuto/1/azure-mgmt-testservice
git	Checkout sdk folders from sdkGen and commit: sdk/testservice/azure-mgmt-testservice
git	Push sdkAuto/1/azure-mgmt-testservice to <org>/<sdk_repo>
section	Call legacy build for azure-sdk-for-python
command	pip install -e ./tools/azure-sdk-tools/
info	Config: {\\"exitCode\\":{\\"showInComment\\":true,\\"result\\":\\"error\\"},\\"stderr\\":{\\"showInComment\\":true},\\"logPrefix\\":\\"buildConfInit\\"}
command	python -m packaging_tools --build-conf azure-mgmt-testservice
info	Config: {\\"exitCode\\":{\\"showInComment\\":true,\\"result\\":\\"error\\"},\\"stderr\\":{\\"showInComment\\":true},\\"logPrefix\\":\\"buildConf\\"}
cmderr	[buildConf] INFO:packaging_tools:Building template azure-mgmt-testservice
cmderr	[buildConf] INFO:packaging_tools.conf:Build default conf for azure-mgmt-testservice
cmderr	[buildConf] INFO:packaging_tools:Template done azure-mgmt-testservice
command	python ./build_package.py --dest sdk/testservice/azure-mgmt-testservice sdk/testservice/azure-mgmt-testservice
info	Config: {\\"exitCode\\":{\\"showInComment\\":true,\\"result\\":\\"error\\"},\\"stderr\\":{\\"showInComment\\":true},\\"logPrefix\\":\\"buildPackage\\"}
cmderr	[buildPackage] warning: no files found matching '*.py' under directory 'tests'
cmderr	[buildPackage] warning: no files found matching '*.yaml' under directory 'tests'
cmderr	[buildPackage] warning: no files found matching '*.py' under directory 'tests'
cmderr	[buildPackage] warning: no files found matching '*.yaml' under directory 'tests'
command	python ./scripts/dev_setup.py -p azure-mgmt-testservice
info	Config: {\\"exitCode\\":{\\"showInComment\\":true,\\"result\\":\\"warning\\"},\\"stderr\\":{\\"showInComment\\":true},\\"logPrefix\\":\\"changelogSetup\\"}
warn	Script return with result [warning] code [1] signal [null] cwd [<sdk_repo>]: python
command	python -m packaging_tools.code_report azure-mgmt-testservice
info	Config: {\\"exitCode\\":{\\"showInComment\\":true,\\"result\\":\\"warning\\"},\\"stderr\\":{\\"showInComment\\":true},\\"logPrefix\\":\\"changelogReport\\"}
warn	Script return with result [warning] code [1] signal [null] cwd [<sdk_repo>]: python
command	python -m packaging_tools.code_report azure-mgmt-testservice --last-pypi
info	Config: {\\"exitCode\\":{\\"showInComment\\":true,\\"result\\":\\"warning\\"},\\"stderr\\":{\\"showInComment\\":true},\\"logPrefix\\":\\"changelogReportLatest\\"}
warn	Script return with result [warning] code [1] signal [null] cwd [<sdk_repo>]: python
warn	Not exact 2 reports found:
warn	Not generating changelog.
info	changelogScript is not configured
info	Use legacy artifact search option
info	Search artifact with: /\\\\.(whl|zip)$/
info	2 artifact found:
info		sdk/testservice/azure-mgmt-testservice/sdk/testservice/azure-mgmt-testservice/azure-mgmt-testservice-1.0.0.zip
info		sdk/testservice/azure-mgmt-testservice/sdk/testservice/azure-mgmt-testservice/azure_mgmt_testservice-1.0.0-py2.py3-none-any.whl
info	Upload 2 artifact:
info		azure-mgmt-testservice-1.0.0.zip
info		azure_mgmt_testservice-1.0.0-py2.py3-none-any.whl
info	Legacy InstallInstruction
github	Get PR in <org>/<sdk_repo> from <org>:sdkAuto/1/azure-mgmt-testservice to release/v3
github	Create new PR
github	PR created at https://github.com/<org>/<sdk_repo>/pull/1
section	Report status
info	Main status [warning] hasBreakingChange [false]
github	Remove label on pull request: CI-BreakingChange-Python
info	Rendered commentSubTitle: 
info	Rendered commentBody: <ul>
  <li>
    <div>
      <code>⚠️</code>Warning Generate from
      <commit_sha>. SDK Automation <version>
    </div>
    <pre>
warn	Skip initScript due to not configured
command	npx n 12
command	python3 -m venv venv
command	pip install --upgrade setuptools wheel pip
command	pip install pathlib jinja2 msrestazure
command	autorest --version&#x3D;V2 --use&#x3D;@microsoft.azure/autorest.python@4.0.73 --python --multiapi --keep-version-file --no-async --python-sdks-folder&#x3D;<work_dir>/<sdk_repo>/sdk ../<spec_repo>/specification/testservice/resource-manager/readme.md</pre
    >
  </li>
  <li>
    <div>
      <span
        ><code>⚠️</code><b>azure-mgmt-testservice</b> [<a
          href=\\"https://sdkautotest/dl/<org>/1/azure-sdk-for-python/azure-mgmt-testservice/logs.txt\\"
          >View full logs</a
        >]&nbsp; [<a
          href=\\"https://github.com/<org>/<sdk_repo>/pull/1\\"
          >Preview SDK Changes</a
        >]
      </span>
    </div>
    <ul>
      <li>
        <a
          href=\\"https://sdkautotest/dl/<org>/1/azure-sdk-for-python/azure-mgmt-testservice/azure-mgmt-testservice-1.0.0.zip\\"
          >azure-mgmt-testservice-1.0.0.zip</a
        >
      </li>
      <li>
        <a
          href=\\"https://sdkautotest/dl/<org>/1/azure-sdk-for-python/azure-mgmt-testservice/azure_mgmt_testservice-1.0.0-py2.py3-none-any.whl\\"
          >azure_mgmt_testservice-1.0.0-py2.py3-none-any.whl</a
        >
      </li>
    </ul>
    <pre>
cmderr	[buildConf] INFO:packaging_tools:Building template azure-mgmt-testservice
cmderr	[buildConf] INFO:packaging_tools.conf:Build default conf for azure-mgmt-testservice
cmderr	[buildConf] INFO:packaging_tools:Template done azure-mgmt-testservice
cmderr	[buildPackage] warning: no files found matching &#x27;*.py&#x27; under directory &#x27;tests&#x27;
cmderr	[buildPackage] warning: no files found matching &#x27;*.yaml&#x27; under directory &#x27;tests&#x27;
cmderr	[buildPackage] warning: no files found matching &#x27;*.py&#x27; under directory &#x27;tests&#x27;
cmderr	[buildPackage] warning: no files found matching &#x27;*.yaml&#x27; under directory &#x27;tests&#x27;
warn	Script return with result [warning] code [1] signal [null] cwd [<sdk_repo>]: python
warn	Script return with result [warning] code [1] signal [null] cwd [<sdk_repo>]: python
warn	Script return with result [warning] code [1] signal [null] cwd [<sdk_repo>]: python
warn	Not exact 2 reports found:
warn	Not generating changelog.</pre
    >
  </li>
</ul>

info	Writing unified pipeline message to pipe.log"
`;
